alphagov/gsp-teams
# Reusable cluster deployer pipeline

## What

Adds a reusable deployer pipeline and corresponding terraform template that can be used to `fly set-pipeline` and spin up a cluster

This is a step towards us managing clusters soley from concourse.

## How to test

* Check out this branch
* Push a pipeline that will provision a cluster in sandbox account called $MY_CLUSTER to the concourse with fly like:
   ```
    fly -t cd-gsp set-pipeline -p sandbox --config pipelines/deployer.yaml --var account-name=sandbox --var account-role-arn=arn:aws:iam::011571571136:role/deployer --var cluster-name=$MY_CLUSTER --yaml-var trusted-developer-keys="$(yq . ./users/*.yaml | jq -s '[ .[].pub ]')" --var splunk_hec_token="NOTATOKEN" --var github-client-secret=NOTASECRET --var github-client-id=NOTID --var gsp-version=eks-firebreak --check-creds
    ```
* Unpause / run the pipeline
* Wait till it's green
* Request your kubeconfig like:
   ```
   aws-vault exec admin@sandbox -- aws eks update-kubeconfig --name $MY_CLUSTER --kubeconfig ./kubeconfig
    ```
* Set your kubeconfig:
    ```
    export KUBECONFIG=$(pwd)/kubeconfig
    ```
* play with your gsp cluster:
    ```
    aws-vault exec admin@sandbox -- kubectl get po --all-namepaces
    ```

##### Note

* should remove references to `eks-firebreak` branch before merging
* @chrisfarms re-wrote this PR text... after continuing the work @blairboy362 started on a pipeline for the EKS branch of GSP ... apologies for editing words into your face
